$NetBSD: patch-ag,v 1.3 2015/12/29 23:34:43 dholland Exp $

Use standard headers.
Don't redeclare standard functions.
Fix yacc syntax errors.
Use my_strtoi here too (patch-af renamed it in ccomp; NetBSD libc has a strtoi).

--- pascomp/bapascal.y.orig
+++ pascomp/bapascal.y
@@ -3,6 +3,7 @@
        /* based on the Pascal grammar from the      */
        /* primost.cs.wisc.edu compiler archive      */
 #include <stdio.h>
+#include <stdlib.h>
 #include <string.h>
 #include <time.h>
 
@@ -25,12 +26,8 @@
 #include "../include/writetab.h"
 #include "../include/pcode.h"
 
-extern   int fprintf(FILE*,const char*,...);
-extern   int printf(const char*,...);
-extern   int free(void*);
 extern   void yyerror(char*);
 extern   int yylex(void); 
-extern   int atoi(char*);
 
 void process_var_inits(int level);
 void add_var_init(int* curr_var, int tix, int level, int value);
@@ -362,7 +359,7 @@
       ;
 
 unsigned_num   : UNSIGNED_INT        /* no reals */
-      {$$ = strtoi(numbertext,numberbase);}
+      {$$ = my_strtoi(numbertext,numberbase);}
       ;
 
 type_dcl_part  : TYPE type_defs ';'
@@ -866,7 +863,6 @@
          { gen_exprval($4); emit1(JZER,$1); free_expr($4); }
       | for_header statement
          { emit1(ENDFOR,1+$1); code[$1].y = lc; }
-      ;
       | cbegin statements CEND
          { if (level == 1) emit(COEND); in_cobegin = 0; }
       ;
@@ -885,6 +881,7 @@
 
 the_while   : WHILE
       { $$ = lc; }
+	;
 
 while_expr  : expr
       { gen_exprval($1); free_expr($1); $$ = lc; emit(JZER);}
